src/Microsoft.XmlSerializer.Generator/src/Sgen.cs

```diff
@@ -232,23 +232,36 @@ private void GenerateFile(List<string> typeNames, string assemblyName, bool prox
             {
                 Type type = types[i];
 
-                if (type != null)
+                try
                 {
-                    bool isObsolete = false;
-                    object[] obsoleteAttributes = type.GetCustomAttributes(typeof(ObsoleteAttribute), false);
-                    foreach (object attribute in obsoleteAttributes)
+                    if (type != null)
                     {
-                        if (((ObsoleteAttribute)attribute).IsError)
+                        bool isObsolete = false;
+                        object[] obsoleteAttributes = type.GetCustomAttributes(typeof(ObsoleteAttribute), false);
+                        foreach (object attribute in obsoleteAttributes)
                         {
-                            isObsolete = true;
-                            break;
+                            if (((ObsoleteAttribute)attribute).IsError)
+                            {
+                                isObsolete = true;
+                                break;
+                            }
                         }
-                    }
 
-                    if (isObsolete)
+                        if (isObsolete)
+                        {
+                            continue;
+                        }
+                    }
+                }
+                catch (FileNotFoundException e)
+                {
+                    if (warnings)
```

Comment:
In `ImportType`, we check `verbose` but here we check `warnings`. Which is correct?